src/plugins/fs/xfs.c

```diff
@@ -227,6 +227,37 @@ gboolean bd_fs_xfs_set_label (const gchar *device, const gchar *label, GError **
     return bd_utils_exec_and_report_error (args, NULL, error);
 }
 
+/**
+ * bd_fs_xfs_check_label:
+ * @label: label to check
+ * @error: (out) (allow-none): place to store error
+ *
+ * Returns: whether @label is a valid label for the xfs file system or not
+ *          (reason is provided in @error)
+ *
+ * Tech category: always available
+ */
+gboolean bd_fs_xfs_check_label (const gchar *label, GError **error) {
+    size_t len = 0;
+
+    len = strlen (label);
+    if (len > 12) {
+        g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_LABEL_INVALID,
+                     "Label for XFS filesystem must be at most 12 characters long.");
+        return FALSE;
+    }
+
+    for (size_t i = 0; i < len; i++) {
```

Comment:
You may also use `strchr()` instead of the cycle...

Reply:
Thanks, I completely forgot that `strchr` exists... Fixed.